purchase_requisition_sale_price/models/purchase_requisition_line.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# (c) 2017 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, fields, api
+from openerp.addons import decimal_precision as dp
+
+
+class PurchaseRequisitionLine(models.Model):
+    _inherit = 'purchase.requisition.line'
+
+    @api.multi
+    @api.depends('product_qty', 'psp_unit')
+    def _compute_psp_subtotal(self):
+        for line in self:
+            line.psp_subtotal = line.psp_unit * line.product_qty
+
+    @api.onchange('margin')
+    def onchange_margin(self):
+        for line in self.filtered(lambda x: x.margin):
```

Comment:
@alfredoavanzosc  filtralo todo en uno `for line in self.filtered(lambda x: x.margin and x.unit_cost):`
Así te queda mucho más limpio y en una sola línea.